### PR TITLE
Tonyvanswet/ele 1368 set up solar and wind layer

### PR DESCRIPTION
**Pull request by tonypls**
## Issue
https://linear.app/electricitymaps/issue/ELE-1368/set-up-solar-and-wind-layer

## Description

It kinda works, at the moment it doesn't show up until you load the page, save vscode then toggle then wind layer. No idea why and have tried debugging with no avail. Maybe something to do with Vite and hot reload + old code

### CODE REVIEW

**Comment by VIKTORVAV99** [comment]
Finally no more moment! 🎉

**Comment by madsnedergaard** [comment]
https://user-images.githubusercontent.com/3296643/208188339-8abb7471-f7bc-48c5-be7c-89c0b37a33bc.mp4

File: web/src/api/getWeatherData.ts [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    parameterNumber: number;
+    refTime: string;
+  };
+}
+export type GfsForecastResponse = [ForecastEntry, ForecastEntry];
+
+const GFS_STEP_ORIGIN = 6; // hours
+const GFS_STEP_HORIZON = 1; // hours
+
+function getForecastStartTime(now: Date) {
```
Comment (madsnedergaard) [inline web/src/api/getWeatherData.ts:28]:
```suggestion
// Data is bucketed into groups for every six hours
function getForecastStartTime(now: Date) {
```

File: web/src/features/weather-layers/wind-layer/scales.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import { scaleLinear, scaleQuantize } from 'd3-scale';
```
Comment (Kongkille) [inline web/src/features/weather-layers/wind-layer/scales.ts:1]: We already have a scales files in util :)

**Review by Kongkille — APPROVED** [review]

File: web/src/features/map/Map.tsx [hunk 3]
```diff
@@ -35,10 +32,20 @@ interface Feature {
 // TODO: Selected feature-id should be stored in a global state instead (and as zoneId).
 // We could even consider not changing it hear, but always reading it from the path parameter?
 export default function MapPage(): ReactElement {
+  // const [viewPortState, setViewPortState] = useState({
```
Comment (tonypls) [inline web/src/features/map/Map.tsx:35]: ✂️ ?

File: web/src/features/weather-layers/grib.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import { addHours } from 'date-fns';
+
+export function getRefTime(grib) {
```
Comment (tonypls) [inline web/src/features/weather-layers/grib.ts:3]: Could there be a comment about what a grib is?

File: web/src/utils/viewport.ts [hunk 5]
```diff
@@ -26,9 +26,6 @@ export function useRefWidthHeightObserver(offsetX = 0, offsetY = 0) {
       }, 0);
       // Update container width on every resize
       window.addEventListener('resize', update);
-      return () => {
-        window.removeEventListener('resize', update);
```
Comment (tonypls) [inline web/src/utils/viewport.ts:30]: Should this be kept?

File: web/src/features/weather-layers/wind-layer/scales.ts (hunk 2 above)
Comment (madsnedergaard) [reply web/src/features/weather-layers/wind-layer/scales.ts:1]: Good catch, I'm working on the solar layer right now and will address this in a new PR :)

File: web/src/utils/viewport.ts (hunk 5 above)
Comment (madsnedergaard) [reply web/src/utils/viewport.ts:30]: Nope, we can't use it on a `useCallback` hook - it's throwing errors in the console :)

File: web/src/features/weather-layers/grib.ts (hunk 4 above)
Comment (madsnedergaard) [reply web/src/features/weather-layers/grib.ts:3]: Eeeeh, I have no idea either - can't come up with a better name right now either, so I suggest we keep this and deal with it at another time 😅

**Review by tonypls — COMMENTED** [review]
LGTM but I can't approve my own PR 😆

**Pull request merged**